repository_service_tuf/cli/generate_key.py

```diff
@@ -0,0 +1,47 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+from rich import prompt
+from rich.console import Console  # type: ignore
+from securesystemslib.interface import (  # type: ignore
+    generate_and_write_ed25519_keypair_with_prompt,
+    generate_and_write_ecdsa_keypair_with_prompt,
+    generate_and_write_rsa_keypair_with_prompt,
+)
+
+from repository_service_tuf.cli import rstuf
+from repository_service_tuf.enums import KeyType
+
+
+console = Console()
+
+
+@rstuf.command()
+def generate_key() -> None:
+    """Generate cryptographic keys using the `securesystemslib` library"""
+
+    key_type: str = prompt.Prompt.ask(
+        f"\nChoose key type",
+        choices=KeyType.get_all_members(),
+        default=KeyType.KEY_TYPE_ED25519.value,
+    )
+
+    filepath: str = prompt.Prompt.ask(
+        "Enter the keys' [green]filename[/] (default path is the current "
+        "working directory) to store the files",
+        default="./rstuf-key",
+    )
```

Comment:
In the implementation, the default path is not the current working directory, it actually defines a filename "rstuf-key".
I suggest only asking the user for the filename and giving the default filename as  `id_{key_type}` following the same standard as `ssh-keygen`. ;)

Additionally, I would suggest an improvement in the user experience, checking if the key already exists and asking if want to overwrite (same as `ssh-keygen`).


```suggestion
    filepath: str = prompt.Prompt.ask(
        "Enter the key's [green]filename[/]:",
        default=f"id_{key_type}",
    )
    if os.path.isfile(filepath) is True:
        overwrite = prompt.Confirm.ask(
            f"Do you want [red]overwrite[/] the existent '{filepath}' file?"
        )
        if overwrite is False:
            raise click.ClickException("Key creation aborted.")    
```

Reply:
I like the suggestion! It is much clearer now as well

Reply:
My option would be `id` as I suggested above, just to be standard. As keys are linked to identification (people, roles, etc.).
But it is just my taste :)